File: data/derived_cot/rq3_prompting/results/translation/r1/cpp_to_py/code_output/BankAccount.py
class BankAccount:
    def __init__(self, balance=0.0):
        self._balance = balance

    def deposit(self, amount):
        if amount < 0:
            raise ValueError("Invalid amount")
        self._balance += amount
        return self._balance

    def withdraw(self, amount):
        if amount < 0:
            raise ValueError("Invalid amount")
        if amount > self._balance:
            raise ValueError("Insufficient balance.")
        self._balance -= amount
        return self._balance

    def view_balance(self):
        return self._balance

    def transfer(self, otherAccount, amount):
        self.withdraw(amount)
        otherAccount.deposit(amount)

File: data/derived_cot/rq1_traces/translation/qwen/java_to_cpp/code_output/BankAccount.cpp
#include <stdexcept>

class BankAccount {
private:
    int balance;

public:
    // Constructor with default parameter for backward compatibility
    BankAccount(int balance = 0) : balance(balance) {}

    int deposit(int amount) {
        if (amount < 0) {
            throw std::invalid_argument("Invalid amount");
        }
        balance += amount;
        return balance;
    }

    int withdraw(int amount) {
        if (amount < 0) {
            throw std::invalid_argument("Invalid amount");
        }
        if (amount > balance) {
            throw std::invalid_argument("Insufficient balance.");
        }
        balance -= amount;
        return balance;
    }

    int viewBalance() const {
        return balance;
    }

    void transfer(BankAccount& otherAccount, int amount) {
        withdraw(amount);
        otherAccount.deposit(amount);
    }
};

int main() {
    BankAccount account1;
    BankAccount account2;
    account1.deposit(1000);
    account1.transfer(account2, 300);
    std::cout << "account1.balance = " << account1.viewBalance() << std::endl;
    std::cout << "account2.balance = " << account2.viewBalance() << std::endl;
    return 0;
}